tests/kernel/mem_protect/stackprot/src/main.c

```diff
@@ -70,6 +70,22 @@ void __attribute__((noinline)) check_input(const char *name, const char *input)
 	TC_PRINT("%s: %s\n", name, buf);
 }
 
+/**
+ *
+ * check_input_fail
+ *
+ * This function is similar to check_input() but does not use an array
+ * so, unless it is built with stack protection for all functions it should fail
+ * with an error different from K_ERR_STACK_CHK_FAIL.
+ */
+void __attribute__((noinline)) check_input_fail(const char *name, const char *input)
```

Comment:
If I remember correctly, there seems to be an `__noinline`.

Reply:
Just followed what was already done in this test. I can add a commit changing the other function too.

Reply:
We don't have it well defined. Just added here https://github.com/zephyrproject-rtos/zephyr/pull/82184